workos/organizations.py

```diff
@@ -45,24 +47,109 @@ def list_organizations(
         Returns:
             dict: Organizations response from WorkOS.
         """
+        warn(
+            "The 'list_organizations' method is deprecated. Please use 'list_organizations_v2' instead.",
+            DeprecationWarning,
+        )
+        if limit is None:
+            limit = RESPONSE_LIMIT
+            default_limit = True
+
         params = {
             "domains": domains,
             "limit": limit,
             "before": before,
             "after": after,
             "order": order,
         }
+
         if order is not None:
-            if not isinstance(order, Order):
-                raise ValueError("'order' must be of asc or desc order")
-            params["order"] = str(order.value)
-        return self.request_helper.request(
+            if isinstance(order, Order):
+                params["order"] = str(order.value)
+
+            elif order == "asc" or order == "desc":
+                params["order"] = order
+            else:
+                raise ValueError("Parameter order must be of enum type Order")
+
+        response = self.request_helper.request(
+            ORGANIZATIONS_PATH,
+            method=REQUEST_METHOD_GET,
+            params=params,
+            token=workos.api_key,
+        )
+
+        response["metadata"] = {
+            "params": params,
+            "method": Organizations.list_organizations,
+        }
+
+        if "default_limit" in locals():
+            response["metadata"]["params"]["default_limit"] = default_limit
+
+        return response
+
+    def list_organizations_v2(
```

Comment:
How does `list_organizations_v2` differ from the current `list_organizations`?

And what's our strategy for migrating off of the old `list_organizations`?

Reply:
The difference is that the v2 method is returning an instance of the class instead of a dictionary. The thought is that it is a better developer experience to call auto pagination directly on the return variable, ex: list_return.auto_paging_iter() which would not be possible on a dictionary. 

As far as a plan to migrate off the old version, we can support both versions until the next major release, at which point the original version would be removed and we'd only support the new version.